src/Compilers/CSharp/Portable/Symbols/Source/SourceNamespaceSymbol.cs

```diff
@@ -372,13 +372,35 @@ private static void CheckMembers(NamespaceSymbol @namespace, Dictionary<string,
 
                     if ((object)other != null)
                     {
-                        if ((nts as SourceNamedTypeSymbol)?.IsPartial == true && (other as SourceNamedTypeSymbol)?.IsPartial == true)
+                        switch (nts, other)
                         {
-                            diagnostics.Add(ErrorCode.ERR_PartialTypeKindConflict, symbol.Locations.FirstOrNone(), symbol);
+                            case (SourceNamedTypeSymbol left, SourceNamedTypeSymbol right) when isLeftAFileTypeInSeparateSourceTreeFromRight(left, right) || isLeftAFileTypeInSeparateSourceTreeFromRight(right, left):
+                                // no error
+                                break;
+                            case (SourceNamedTypeSymbol { IsPartial: true }, SourceNamedTypeSymbol { IsPartial: true }):
+                                diagnostics.Add(ErrorCode.ERR_PartialTypeKindConflict, symbol.Locations.FirstOrNone(), symbol);
+                                break;
+                            default:
+                                diagnostics.Add(ErrorCode.ERR_DuplicateNameInNS, symbol.Locations.FirstOrNone(), name, @namespace);
+                                break;
                         }
-                        else
+
+                        // prototype(ft): better name? better diagnostic?
```

Comment:
>prototype [](http://example.com/codeflow?start=27&length=9)

I think PROTOTYPE must be upper-case to trigger the check: `$prototypes = Get-ChildItem -Path src, eng, scripts -Exclude *.dll,*.exe,*.pdb,*.xlf,todo-check.ps1 -Recurse | Select-String -Pattern 'PROTOTYPE' -CaseSensitive -SimpleMatch`

Reply:
thanks, I'll try to fix misuses of prototype.